### PR TITLE
Use 'subclause', not 'section'.

### LINKED ISSUE #307: Inconsistent use of "in this section" vs "in this sub-clause"
"In this section" is far more common, but we have the following mentions of sub-clauses:

[expr.const]
[meta.unary]
[meta.unary.prop.query]
[meta.rel]
[meta.trans]
[meta.numerics]
[rand.eng]
[futures.state]

### PR DESCRIPTION
**Pull request by jensmaurer**
The ISO Directives, Part 2, prescribe the term 'subclause'.

Fixes #307.

### CODE REVIEW

File: source/back.tex [hunk 1]
```diff
@@ -22,8 +22,8 @@ \chapter{Bibliography}
 
 \clearpage
 \renewcommand{\glossaryname}{Cross references}
-\renewcommand{\preglossaryhook}{This annex lists each section label and the
-corresponding section number and page number, in alphabetical order by label.\\}
+\renewcommand{\preglossaryhook}{This annex lists each subclause label and the
+corresponding subclause number and page number, in alphabetical order by label.\\}
```
Comment (zygoloid) [inline source/back.tex]: These ones mean "clause or subclause". We could use "[...] lists each clause or subclause label and [...]" but perhaps there's a better way to say that?

File: source/back.tex [hunk 2]
```diff
@@ -34,7 +34,7 @@ \chapter{Bibliography}
 \clearpage
 \input{xrefdelta}
 \renewcommand{\glossaryname}{Cross references from ISO \CppXIV}
-\renewcommand{\preglossaryhook}{All section labels from
+\renewcommand{\preglossaryhook}{All subclause labels from
```
Comment (zygoloid) [inline source/back.tex]: Likewise, even though we haven't removed any Clause labels yet.

File: source/back.tex [hunk 3]
```diff
@@ -71,7 +71,7 @@ \chapter{Bibliography}
 }
 
 \clearpage
-\renewcommand{\preindexhook}{The entries in this section are rough descriptions; exact
+\renewcommand{\preindexhook}{The entries in this subclause are rough descriptions; exact
```
Comment (zygoloid) [inline source/back.tex]: This is not a subclause, it's an index.

File: source/back.tex (hunk 1 above)
Comment (jensmaurer) [inline source/back.tex]: I'm going for "clause or subclause" for now.

File: source/back.tex (hunk 3 above)
Comment (jensmaurer) [inline source/back.tex]: Fixed.

**Comment by jensmaurer** [comment]
Rebased and addressed review comments.

**Comment by tkoeppe** [comment]
Just a random observation, because I was also working on this: in lib-intro.tex, the proposed change is problematic, because we currently say for section `X`:

> This section describes Foo. Subclause `X.1` describes meta-Foos, subclause `X.2` describes para-Foos.

If you change the first "section" to "subclause", then the hierarchy is lost.

**Comment by jensmaurer** [comment]
@tkoeppe: lib-intro.tex is rather inconsistent here. My suggestion (implemented in the patch) is to remove the "overview" sub-headings in these cases so that the overview text is at the right level. Yes, it's a "hanging paragraph" in ISO speak, but strictly speaking, these introductory sentences should not be shown as normative text anyway. Do we want to find a typographical convention other than "[ Note: ... ]" that keeps such texts readable, but makes them non-normative? Maybe some indent? Or a non-serif font? Both? (Oh, we wanted a non-serif font for the grammar.)

File: source/diagnostics.tex [hunk 4]
```diff
@@ -939,7 +939,7 @@
 Classes may be derived from \tcode{error_category} to support
 categories of errors in addition to those defined in this document.
 Such classes shall behave as specified in this
-subclause. \begin{note} \tcode{error_category} objects are
+subclause~\ref{syserr.errcat}. \begin{note} \tcode{error_category} objects are
```
Comment (zygoloid) [inline source/diagnostics.tex]: Use `\iref` for this inline reference.

File: source/lib-intro.tex [hunk 5]
```diff
@@ -1324,10 +1324,8 @@
 
 \rSec2[using]{Using the library}
 
-\rSec3[using.overview]{Overview}
-
 \pnum
-This section describes how a \Cpp program gains access to the facilities of the
+This subclause describes how a \Cpp program gains access to the facilities of the
```
Comment (zygoloid) [inline source/lib-intro.tex]: How about: keep the `Overview` subclause, and use "Subclause \ref{using} describes" here.

File: source/macros.tex [hunk 6]
```diff
@@ -62,7 +62,7 @@
 
 %%--------------------------------------------------
 %% Sectioning macros.
-% Each section has a depth, an automatically generated section
+% Each subclause has a depth, an automatically generated subclause
```
Comment (zygoloid) [inline source/macros.tex]: I don't think this change is right. :)

Comment (jensmaurer) [reply source/macros.tex]: There didn't show up in diffpdf.

File: source/lib-intro.tex (hunk 5 above)
Comment (jensmaurer) [reply source/lib-intro.tex]: Fixed.

File: source/xrefdelta.tex [hunk 7]
```diff
@@ -36,5 +36,10 @@
   \secref{fs.path.generic},
   \secref{fs.race.behavior}}
 
+% Separate sections for introductory paragraphs were removed.
+\removedxref{using.overview}
+\removedxref{constraints.overview}
+\removedxref{conforming.overview}
```
Comment (zygoloid) [inline source/xrefdelta.tex]: I think this is now redundant.

Comment (jensmaurer) [reply source/xrefdelta.tex]: Indeed. Fixed.

**Comment by jensmaurer** [comment]
Change to [xrefdelta] reverted, as advised.

**Pull request merged**